server/aesdsocket.c

```diff
@@ -0,0 +1,303 @@
+#include <sys/socket.h>
+#include <sys/types.h>
+#include <sys/socket.h>
+#include <netdb.h>
+#include <arpa/inet.h>
+
+#include <syslog.h>
+#include <unistd.h>
+#include <fcntl.h>
+#include <stdio.h>
+#include <string.h>
+#include <errno.h>
+
+#include <signal.h>
+#include <stdbool.h>
+#include <stdlib.h>
+
+// Macros
+#define PORT_NUM_STR "9000"
+#define MAX_PENDING_CONNECTIONS 3
+#define AESD_SOCKET_DATA_FILE "/var/tmp/aesdsocketdata"
+#define PACKET_BUFFER_SIZE 512
+
+// Global variables
+bool terminate = false;
+bool connectionInProgress = false;
+struct addrinfo* serverInfo = NULL;
+int serverSockfd = -1;
+int acceptedSocketfd = -1;
+char* pktBuf = NULL;
+
+/*
+ * Function for handling registered signals
+ */
+void SignalHandler(int signo)
+{
+	syslog(LOG_INFO, "Caught signal, exiting");
+	terminate = true;
+
+	// No active connections, exit
+	if(!connectionInProgress)
+	{
+		int rc = remove(AESD_SOCKET_DATA_FILE);
+		if(rc == -1)
+			perror("remove socket data file");
+		if(pktBuf != NULL)
+			free(pktBuf);
+		if(acceptedSocketfd != -1)
+			close(acceptedSocketfd);
+		if(serverInfo != NULL)
+			freeaddrinfo(serverInfo);
+		if(serverSockfd != -1)
+			close(serverSockfd);
+
+		exit(0);
+	}
+}
+
+/*
+ * Function to send file content through socket
+ */
+void SendFileDataToClient(int fileFd, int socketFd)
+{
+	ssize_t ret;
+	char buf[512];
+
+	// Seek to start of file
+	lseek(fileFd, 0, SEEK_SET);
+
+	while ((ret = read (fileFd, buf, sizeof buf)) != 0)
+	{
+        if (ret == -1) 
```

Comment:
Good approach to send here.
Check your spaces/tabs here though, something isn't formatted properly.